adr/minimal-sdk.md

```diff
@@ -0,0 +1,53 @@
+
+This two-pronged approach (a minimal "first-step" SDK and a clear "next-step" recommendation of powerful external tools) provides the best balance of speed, user choice, and focused engineering effort.
+
+## 3. Consequences
+
+### Positive
+
+* **Rapid Time-to-Market:** A minimal SDK is a fast way to deliver a functional agent construct to our users, allowing us to provide a complete end-to-end story quickly.
+* **Empowers User Choice:** This strategy explicitly embraces the rich AI ecosystem. Customers with existing expertise in established frameworks can use their preferred tools, while new users have a simple, non-intimidating entry point.
+* **Low Engineering Overhead:** By strictly limiting the scope of our own SDK, we avoid the massive maintenance burden of creating and supporting a full-featured framework. Engineering efforts are focused and contained.
+* **Simplified Tooling:** Having a single, simple, Red Hat-owned SDK as a target greatly simplifies the development of any integrated tooling (e.g., agent creation wizards in the RHOAI console).
+* **Avoids Containerization Complexity:** Choosing an SDK over a container image avoids the overhead of building, distributing, securing, and running a container for what is fundamentally client-side logic, simplifying the developer workflow.
+
+### Negative
+
+* **Weakens the "Unified Framework" Narrative:** This approach makes it harder to market Llama Stack or a Red Hat offering as a single, all-in-one agent framework. The story is more nuanced ("use our simple SDK to start, then use these other tools").
+* **Risk of SDK Fragmentation and Confusion:** We must provide extremely clear documentation and guidance to help users understand when to use the minimal Red Hat SDK and when it's time to "graduate" to a more powerful framework.
+* **Limited to Python:** The decision to write the SDK only in Python will limit its direct usefulness for customers who prefer other languages. It also presents a challenge for our internal tooling teams who may use other languages like Go. We accept these trade-offs, acknowledging that a majority of the target AI/ML developer audience works primarily in Python and that workarounds can be found for internal tooling.
+* **Potential for an Inadequate SDK:** There is a risk that our minimal SDK will be too limited even for slightly complex use cases, forcing a migration to another framework very early in the development process. The initial scope must be defined carefully to be useful.
+* **Implied Support Burden:** By officially recommending third-party frameworks, we create a customer expectation of support. We will need to clearly define our support boundaries and invest resources in creating compatibility documentation, examples, and potentially addressing integration-related issues.
```

Comment:
Even if we clearly specify that we would not support the framework they are using, i expect we may still have to spend time and resources investigating reports to determine if a reported issue's cause really lies in their framework or on the server side to know if any action is required on our part

Reply:
Yes, that's a reasonable point.  I will adjust the text to note that.